stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -327,6 +327,11 @@ public void request(int count) {
     public void setMessageCompression(boolean enable) {
       call.setMessageCompression(enable);
     }
+
+    @Override
+    public void cancel() {
+      call.cancel("Call cancelled by request.", new StatusException(Status.CANCELLED));
```

Comment:
FYI: this exception will become the _cause_ of the Status, so this would produce a weird Status. We basically do the equivalent of `Status.CANCELLED.withDescription(passedString).withCause(passedThrowable)`.

And _yes_, that's really surprising given server-side usage of `onError()`, but with the old zero-arg cancel() we would throw the `Throwable` away.